src/basic_recipes/poly.jl

```diff
@@ -90,7 +90,9 @@ function poly_convert(polygon::Polygon, transform_func=identity)
     outer = metafree(coordinates(polygon.exterior))
     # TODO consider applying f32 convert here too. We would need to identify this though...
     PT = float_type(outer)
-    points = Vector{PT}[apply_transform(transform_func, outer)]
+    # Note that this should not be coerced to be a `Vector{PT}`,
+    # since `apply_transform` can change points from e.g 2D to 3D.
+    points = [apply_transform(transform_func, outer)]
     points_flat = PT[outer;]
     for inner in polygon.interiors
```

Comment:
Some tests would be nice.
What about the `PT[outer;]`?

Reply:
That is fine, since points_flat contains the original points.

Will add some tests!